File: backend/app/core/upgrades/stat_modifiers/player_health_4.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerHealth4(UpgradeImplementation):
    id = 'player_health_4'
    name = 'Reinforced Core'
    description = '+130 max health'
    rarity = 'epic'
    type = 'stat_modifier'
    cost = 20

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 130
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/player_health_4.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerHealth4Def: UpgradeDef = {
  id: "player_health_4",
  name: "Reinforced Core",
  description: "+130 max health",
  rarity: RarityID.Epic,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 20,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: 130,
  stackable: true,
  maxStacks: 99999999,
}

export class PlayerHealth4 extends Upgrade {}
